be/src/storage/compaction.cpp

```diff
@@ -245,6 +245,10 @@ Status Compaction::_merge_rowsets_vertically(size_t segment_iterator_num, Statis
     TRACE_COUNTER_SCOPE_LATENCY_US("merge_rowsets_latency_us");
     auto mask_buffer = std::make_unique<RowSourceMaskBuffer>(_tablet->tablet_id(), _tablet->data_dir()->path());
     auto source_masks = std::make_unique<std::vector<RowSourceMask>>();
+    // ensure all input rowsets are loaded into memory
+    for (auto& rowset : _input_rowsets) {
+        RETURN_IF_ERROR(rowset->load());
```

Comment:
Why can this operation assure that rowset won't be released?
I think other thread can release these rowsets after loading()